### PR TITLE
Match use_tweester, and munchlesia + misc partners

### PR DESCRIPTION
**Pull request by ThreeMustyFears**

### CODE REVIEW

**Comment by BowserSlug** [comment]
⚠️ This PR introduces 39 warnings:
- src/world/action/../../npc.h:101: warning: function declaration isn't a prototype
- src/world/action/../../npc.h:103: warning: function declaration isn't a prototype
- src/world/action/../../npc.h:185: warning: function declaration isn't a prototype
- src/world/action/../../npc.h:189: warning: function declaration isn't a prototype
- src/world/action/../../npc.h:210: warning: redundant redeclaration of `npc_dyn_entity_draw_no_op' in same scope
- src/world/action/../../npc.h:259: warning: redundant redeclaration of `get_enemy' in same scope
- src/world/action/../../npc.h:35: warning: redundant redeclaration of `get_npc_by_index' in same scope
- src/world/action/../../npc.h:51: warning: function declaration isn't a prototype
- src/world/action/../../npc.h:58: warning: redundant redeclaration of `npc_move_heading' in same scope
- src/world/action/../../npc.h:60: warning: redundant redeclaration of `get_npc_unsafe' in same scope
- src/world/action/../../npc.h:63: warning: redundant redeclaration of `get_npc_safe' in same scope
- src/world/action/../../npc.h:81: warning: redundant redeclaration of `set_npc_yaw' in same scope
- src/world/action/../../npc.h:97: warning: function declaration isn't a prototype
- src/world/action/../../npc.h:99: warning: function declaration isn't a prototype
- src/world/action/../partners.h:29: warning: redundant redeclaration of `D_8010C954' in same scope
- src/world/action/use_munchlesia.c:36: warning: passing arg 1 of `add_vec2D_polar' from incompatible pointer type
- src/world/action/use_munchlesia.c:47: warning: passing arg 1 of `add_vec2D_polar' from incompatible pointer type
- src/world/action/use_tweester.c:24: warning: assignment from incompatible pointer type
- src/world/action/use_tweester.c:32: warning: passing arg 1 of `mem_clear' from incompatible pointer type
- src/world/partner/bombette.c:121: warning: initialization makes integer from pointer without a cast
- src/world/partner/bombette.c:124: warning: implicit declaration of function `partner_init_put_away'
- src/world/partner/bombette.c:126: warning: implicit declaration of function `partner_put_away'
- src/world/partner/bombette.c:16: warning: implicit declaration of function `partner_init_get_out'
- src/world/partner/bombette.c:18: warning: implicit declaration of function `partner_get_out'
- src/world/partner/bombette.c:32: warning: passing arg 1 of `mem_clear' from incompatible pointer type
- src/world/partner/bombette.c:37: warning: assignment makes pointer from integer without a cast
- src/world/partner/bow.c:35: warning: passing arg 1 of `mem_clear' from incompatible pointer type
- src/world/partner/bow.c:38: warning: assignment makes pointer from integer without a cast
- src/world/partner/kooper.c:26: warning: passing arg 1 of `mem_clear' from incompatible pointer type
- src/world/partner/kooper.c:31: warning: assignment makes pointer from integer without a cast
- src/world/partner/lakilester.c:23: warning: passing arg 1 of `mem_clear' from incompatible pointer type
- src/world/partner/lakilester.c:28: warning: assignment makes pointer from integer without a cast
- src/world/partner/parakarry.c:105: warning: initialization makes integer from pointer without a cast
- src/world/partner/parakarry.c:108: warning: implicit declaration of function `partner_init_put_away'
- src/world/partner/parakarry.c:110: warning: implicit declaration of function `partner_put_away'
- src/world/partner/parakarry.c:11: warning: implicit declaration of function `partner_init_get_out'
- src/world/partner/parakarry.c:13: warning: implicit declaration of function `partner_get_out'
- src/world/partner/parakarry.c:27: warning: passing arg 1 of `mem_clear' from incompatible pointer type
- src/world/partner/parakarry.c:31: warning: assignment makes pointer from integer without a cast
- src/world/partner/sushie.c:31: warning: passing arg 1 of `mem_clear' from incompatible pointer type
- src/world/partner/sushie.c:35: warning: assignment makes pointer from integer without a cast

File: src/world/action/use_munchlesia.c [hunk 1]
```diff
@@ -1,3 +1,79 @@
...
+            D_802B62D0 = 16.0f;
+            D_802B62D4 = 4.0f;
+            D_802B62D8 = 1.0f;
+            D_802B62D4 = 1.875f;
+            playerStatus->fallState++;
+            D_802B62DC = playerStatus->targetYaw;
+            break;
+        case 1:
+            playerStatus->position.y += D_802B62D0;
+            D_802B62D0 = D_802B62D0 - D_802B62D8;
```
Comment (ethteck) [inline src/world/action/use_munchlesia.c:35]: `D_802B62D0 -= D_802B62D8;`

File: src/world/action/use_munchlesia.c [hunk 2]
```diff
@@ -1,3 +1,79 @@
...
+            gCameras[0].targetPos.x = playerStatus->position.x;
+            gCameras[0].targetPos.y = playerStatus->position.y;
+            gCameras[0].targetPos.z = playerStatus->position.z;
+            if (D_802B62D0 <= 0.0f) {
+                playerStatus->fallState++;
+            }
+            break;
+        case 2:
+            playerStatus->position.y += D_802B62D0;
+            D_802B62D0 = D_802B62D0 - D_802B62D8;
```
Comment (ethteck) [inline src/world/action/use_munchlesia.c:46]: `D_802B62D0 -= D_802B62D8;`

File: src/world/partner/bombette.c [hunk 3]
```diff
@@ -15,7 +18,88 @@ ApiStatus func_802BD300_318050(Evt* script, s32 isInitialCall) {
...
+    }
+    playerData->unk_2F4[3]++;
+    npc->flags |= 0x10000;
+    entity = D_8010C954;
+    if (entity == NULL) {
+        partner_walking_update_player_tracking(npc);
+        partner_walking_update_motion(npc);
+        return 0;
+    }
+    switch (D_802BE89C_3195EC->unk_04) {
```
Comment (ethteck) [inline src/world/partner/bombette.c:43]: empty line before the switch please

**Review by ethteck — CHANGES_REQUESTED** [review]
Looks great! I noticed that this partner func may be duplicated (with the only difference being a bss symbol or something):

```
ethteck@SNAILSHELL:~/repos/papermario$ ./tools/find_duplicates.py func_802BD2D4_320E24
func_802BD2D4_320E24 - found 7 matches total:
0.99 - func_802BD338_318088
0.99 - func_802BD1D0_3170F0
0.99 - func_802BD180_3196F0
0.99 - func_802BD260_31B280
0.99 - func_802BD168_323AB8
0.99 - func_802BD14C_324A5C
0.98 - func_802BF5A0_320310
```

It'd be good to try to de-duplicate it instead of having it in the project a bunch of times. The idea would be to make an inc.c file (like we have in the common/ subdirs) and #include it from the partner C files. To handle the different sym, you'd ideally use N(someName) but this may be kinda annoying / some effort. Let me know if any of this makes sense lol

**Comment by ThreeMustyFears** [comment]
If I understand this right the idea would be to only write the function once in the inc.c file, then include that in place of the function. The N(someName) makes it differ when included in a different partner's c function depending on the defined namespace.

**Comment by ethteck** [comment]
Yep, exactly!

**Comment by ethteck** [comment]
Oh wow, thanks for going the extra mile with de-duplicating that. I wasn't aware there were functional changes besides just changing out a symbol or something... we should wait to see what @nanaian says about it, but it might be preferable to go back to how it was before just to avoid ifdefs and stuff.. sorry about that if so. But we will have to do de-duplication for many other functions, so it's good practice regardless 😁

**Comment by ethteck** [comment]
I talked with alex, and he agrees that it's probably better the way it was before...sorry about that. Once you restore back to your previous commit I'l merge.

**Comment by ThreeMustyFears** [comment]
No worries, it was interesting to figure out anyway

**Pull request merged**